lib/s3_meta_sync/syncer.rb

```diff
@@ -83,6 +87,23 @@ def download(source, destination)
       end
     end
 
+    # Sometimes SIGTERM causes Dir.mktmpdir to not properly clear the temp folder
+    # Remove day old folders
+    def clear_old_temp_folders
+      path = File.join(Dir.tmpdir, TEMP_FOLDER_PREFIX + '*')
+
+      total = Dir.glob(path).map do |dir|
+        day = 60 * 60 * 24
+        FileUtils.rm_rf(dir) if older_than(dir, day)
+      end.compact.count
```

Comment:
semi dangerous in case rm_rf returns nil and it is then ignored due to compact ... but should be good